src/main/java/androidx/app/Activity.kt

```diff
@@ -0,0 +1,19 @@
+package androidx.app
+
+import android.app.Activity
+import android.content.Intent
+import android.os.Bundle
+import android.support.annotation.RequiresApi
+
+/**
+ * Syntactic sugar for [Activity.startActivityForResult(Intent, Int)][Activity.startActivityForResult].
+ */
+inline fun <reified T> Activity.startActivityForResult(code: Int, block: Intent.() -> Unit = {}) =
```

Comment:
It's definitely the first extension that i wanted to add. Actually your reified parameter must be an activity so it should be <reified T : Activity>

Reply:
Same for all other start extensions. In addition, we must handle startActivityForResult for fragments as well. I'm not sure if the support library is in the scope of this library.

Reply:
Agree on both points. I added the upper bound and the extensions for regular fragments. From what I can tell the support library isn't in the scope right now.